petastorm/arrow_reader_worker.py

```diff
@@ -44,29 +45,36 @@ def read_next(self, workers_pool, schema, ngram):
             # Convert arrow table columns into numpy. Strings are handled differently since to_pandas() returns
             # numpy array of dtype=object.
             result_dict = dict()
-            for column in result_table.columns:
+            for column_name, column in compat_table_columns_gen(result_table):
                 # Assume we get only one chunk since reader worker reads one rowgroup at a time
 
                 # `to_pandas` works slower when called on the entire `data` rather directly on a chunk.
-                if result_table.column(0).data.num_chunks == 1:
+                if compat_column_num_chunks(result_table.column(0)) == 1:
                     column_as_pandas = column.data.chunks[0].to_pandas()
                 else:
                     column_as_pandas = column.data.to_pandas()
 
+                # pyarrow < 0.15.0 would always return a numpy array. Starting 0.15 we get pandas series, hence we
+                # convert it into numpy array
+                if isinstance(column_as_pandas, pd.Series):
+                    column_as_numpy = column_as_pandas.as_matrix()
+                else:
+                    column_as_numpy = column_as_pandas
+
                 if pa.types.is_string(column.type):
-                    result_dict[column.name] = column_as_pandas.astype(np.unicode_)
+                    result_dict[column_name] = column_as_numpy.astype(np.unicode_)
                 elif pa.types.is_list(column.type):
                     # Assuming all lists are of the same length, hence we can collate them into a matrix
-                    list_of_lists = column_as_pandas
+                    list_of_lists = column_as_numpy
```

Comment:
Is the switch from pandas to numpy due to list issue in 0.15.0, or is there another reason

Reply:
:+1:  to this question